### ISSUE TITLE
pre-commit install-hooks not installing hadolint binary

### ISSUE THREAD

**Issue by jamesbraza** [hadolint/hadolint#886]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

I added the following to my `.pre-commit-config.yaml` per [`INTEGRATION.md` docs](https://github.com/hadolint/hadolint/blob/master/docs/INTEGRATION.md#pre-commit):

```yaml
  - repo: https://github.com/hadolint/hadolint
    rev: v2.10.0
    hooks:
      - id: hadolint
```

I then ran `pre-commit install-hooks` and expect it to work.

```
[INFO] Initializing environment for https://github.com/psf/black.
[INFO] Initializing environment for https://github.com/hadolint/hadolint.
[INFO] Installing environment for https://github.com/psf/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
```

Note that `hadolint` doesn't show up as being set up.

### Actual behavior

When running `pre-commit`:

```
Lint Dockerfiles..............................................................Failed
- hook id: hadolint
- exit code: 1

Executable `hadolint` not found
```

I would expect `pre-commit install-hooks` to install the `hadolint` binary.

### Steps to reproduce the behavior

Documented above, I just:
1. Added the `pre-commit` section to `.pre-commit-config.yaml`
2. Ran `pre-commit install-hooks`

**Additional environment details (OS, stack version, etc.)**

macOS Big Sur v11.5.1.   Though I don't want to `brew install hadolint`, I would like `pre-commit` to encapsulate it as it does every other hook.

**Comment by matroscoe** [comment]
I can confirm this is also an issue on Ubuntu 22.04 (not surprising but just to confirm)

**Comment by chandrasekharan98** [comment]
While this gets fixed, referring from [this SO answer](https://stackoverflow.com/a/62370018/12864260), with Linux distributions we can also run the following command (upgrade the version as and when necessary)
```console
wget -O /bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.12.0/hadolint-Linux-x86_64
```

**Comment by CameronHudson8** [comment]
Don't forget to make it executable. 🙂 
```
wget -O /bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.12.0/hadolint-Linux-x86_64
chmod +x /bin/hadolint 
```

**Comment by sandstromviktor** [comment]
Any updates? I want to run hadolint as a pre-commit in github actions. Would like to avoid having a additional step of downloading like this.

**Comment by CarrotManMatt** [comment]
> I want to run hadolint as a pre-commit in github actions. Would like to avoid having a additional step of downloading

If you are only running pre-commit in CI (GitHub actions), the `hadolint-docker` pre-commit hook provided by hadolint does not require the download step as all `ubuntu` GitHub action runners come with docker set up ready to use.

Using this hook in your `.pre-commit-config.yaml` will however require all your project's developers to have docker installed in order to run the hook locally, or skip this hook when committing.

**Comment by shenxianpeng** [comment]
For anyone still running into this — I ran into the same issue and ended up building a pre-commit hook that ships the hadolint binary via PyPI, so it gets installed automatically without any manual binary download.

Add the following to your `.pre-commit-config.yaml`:

```yaml
repos:
  - repo: https://github.com/shenxianpeng/hadolint-pre-commit
    rev: v2.14.0.1
    hooks:
      - id: hadolint
```
The binary is distributed as a Python package, so `pre-commit` handles everything — no manual binary downloads or Docker required (if you use `hadolint-docker` hook). Supports Linux, macOS, and Windows.
